doc/guides/getting_started/src/e2e_regression_setup.md

```diff
@@ -0,0 +1,40 @@
+# ROM E2E Regressions setup
```

Comment:
Maintainers and downstream consumers are also interested in understanding how the test setup works so that they can add or modify existing test cases. It would be useful to have a diagram showing all the components (e.g. DUT, opentitanlib, Bazel, etc).

Reply:
Done